tests/infra/consortium.py

```diff
@@ -293,13 +295,31 @@ def retire_node(self, remote_node, node_to_retire):
 
         with remote_node.client() as c:
             r = c.get(f"/node/network/nodes/{node_to_retire.node_id}")
-            # Until we have 2tx-promotion, we expect RETIRING but not RETIRED yet.
-            expected = (
-                NodeStatus.RETIRED.value
-                if self.consensus != "bft"
-                else NodeStatus.RETIRING.value
+            status = r.body.json()["status"]
```

Comment:
From a UX perspective, it seems a little awkward to change the check for successfully retire node proposal logic depending on the type of reconfiguration. In all cases, the node should become `RETIRED` after a relatively short timeout, sometimes passing through the `RETIRING` state?

Reply:
Not in the test scripts - if there are bugs, they may not retire at all, so we need a check with timeout. Also, the other test scripts rely on the fact that this happens pretty much instantaneously and try to re-use the ip/port of the retiring node, so we have to make sure the node is actually gone before we continue (or add waiting loops in the corresponding tests).

Reply:
OK. In this case, can the node already be in `retired` state in this RPC, even with a 2tx reconfiguration (i.e. if node was fast enough to retire itself)? It feels like the logic could be unified: wait for the node to be marked as `RETIRED` within a timeout. If we're in 1tx reconfiguration, the check should pass straight away.